### PR TITLE
closes #347 increase test coverage in srcfolio migration toolsmarc rules transformationrules mapper holdingspy

### LINKED ISSUE #347: Increase test coverage in src/folio_migration_tools/marc_rules_transformation/rules_mapper_holdings.py
# Description
Your task is to create happy path tests for the methods _remove_from_id_map_ and _set_default_call_number_type_if_empty_ in rules_mapper_holdings.py.
You will need to mock RulesMapperHoldings.

# DoD
- [ ] You have read the testing section in https://github.com/FOLIO-FSE/folio_migration_tools/blob/main/CONTRIBUTING.md#testing
- [ ] You have created a branch from this issue and checked it out locally
- [ ] You have run the entire test suite as described in Contributing.md making sure the code works before you start working on it
- [ ] You have written the test outlined in the description above
- [ ] You have successfully run the test
- [ ] You have performed all steps outlined in the [Github workflow](https://github.com/FOLIO-FSE/folio_migration_tools/blob/main/CONTRIBUTING.md#github-workflow) section in the Contributing.md file, including creating a pull request.

### PR DESCRIPTION
**Pull request by branchedelac**
Apologies for the extra commits in this branch -- they were related to issue 218 and had already been merged in.

The real changes are in tests/test_rules_mapper_holdings.py

### CODE REVIEW

File: tests/test_rules_mapper_holdings.py [hunk 1]
```diff
@@ -29,3 +32,44 @@ def test_get_marc_textual_stmt_correct_order_and_not_deduped():
...
+        "h14554914": "c9c44650-11e2-5534-ae50-01a1aa0fbd66",
+    }
+
+    former_ids = ["h15066915"]
+
+    RulesMapperHoldings.remove_from_id_map(mocked_rules_mapper_holdings, former_ids)
+
+    # The ids in the former_ids have been removed, any others are still there
+    assert "h15066915" not in mocked_rules_mapper_holdings.holdings_id_map.keys()
+    assert "h14554914" in mocked_rules_mapper_holdings.holdings_id_map.keys()
```
Comment (fontanka16) [inline tests/test_rules_mapper_holdings.py:50]: I am pretty sure you can ommit the .keys() here, but it certainly makes it clearer.

File: tests/test_rules_mapper_holdings.py [hunk 2]
```diff
@@ -29,3 +32,44 @@ def test_get_marc_textual_stmt_correct_order_and_not_deduped():
...
+def test_remove_from_id_map():
+    mocked_rules_mapper_holdings = Mock(spec=RulesMapperHoldings)
+    mocked_rules_mapper_holdings.holdings_id_map = {
+        "h15066915": "5a0af31f-aa4a-5215-8a60-712b38cd6cb6",
+        "h14554914": "c9c44650-11e2-5534-ae50-01a1aa0fbd66",
+    }
+
+    former_ids = ["h15066915"]
+
+    RulesMapperHoldings.remove_from_id_map(mocked_rules_mapper_holdings, former_ids)
```
Comment (fontanka16) [inline tests/test_rules_mapper_holdings.py:46]: I love this construct!

**Review by fontanka16 — APPROVED** [review]
Beautiful!

File: tests/test_rules_mapper_holdings.py (hunk 1 above)
Comment (branchedelac) [reply tests/test_rules_mapper_holdings.py:50]: That's how I had it from the start -- but my impression was that that was making the tests pass even when they shouldn't and that adding the .keys() fixed this.

**Pull request merged**